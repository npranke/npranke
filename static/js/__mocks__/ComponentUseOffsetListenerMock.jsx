import PropTypes from 'prop-types'
import React from 'react'

import useOffsetListener from '@components/hooks/OffsetListener'

function ComponentUseOffsetListenerMock(props) {
    const { height, width } = useOffsetListener(props.elementRef)

    return (
        <div className="component-use-offset-listener-mock">
            <div className="height">{ height }</div>
            <div className="width">{ width }</div>
        </div>
    )
}

ComponentUseOffsetListenerMock.propTypes = {
    elementRef: PropTypes.shape({
        current: PropTypes.object,
    }).isRequired,
}

export default ComponentUseOffsetListenerMock
