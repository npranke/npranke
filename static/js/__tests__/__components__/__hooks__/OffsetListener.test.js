import { fireEvent, render, screen } from '@testing-library/react'

import ComponentUseOffsetListenerMock from
    '@__mocks__/ComponentUseOffsetListenerMock'

describe('OffsetListener', () => {
    describe('without elementRef.current', () => {
        test('sets initial height to 0', () => {
            render(
                <ComponentUseOffsetListenerMock elementRef={ {} } />,
            )

            expect(
                screen.getByTestId('height-div-elem').textContent,
            ).toEqual('0')
        })

        test('sets initial width to 0', () => {
            render(
                <ComponentUseOffsetListenerMock elementRef={ {} } />,
            )

            expect(
                screen.getByTestId('width-div-elem').textContent,
            ).toEqual('0')
        })

        test('sets height on resize with offset height value', () => {
            const { rerender } = render(
                <ComponentUseOffsetListenerMock elementRef={ {} } />,
            )

            rerender(
                <ComponentUseOffsetListenerMock
                    elementRef={ {
                        current: {
                            offsetHeight: 50,
                            offsetWidth: 75,
                        },
                    } }
                />,
            )

            expect(
                screen.getByTestId('height-div-elem').textContent,
            ).toEqual('0')

            fireEvent(window, new Event('resize'))

            expect(
                screen.getByTestId('height-div-elem').textContent,
            ).toEqual('50')
        })

        test('sets width on resize with offset width value', () => {
            const { rerender } = render(
                <ComponentUseOffsetListenerMock elementRef={ {} } />,
            )

            rerender(
                <ComponentUseOffsetListenerMock
                    elementRef={ {
                        current: {
                            offsetHeight: 50,
                            offsetWidth: 75,
                        },
                    } }
                />,
            )

            expect(
                screen.getByTestId('width-div-elem').textContent,
            ).toEqual('0')

            fireEvent(window, new Event('resize'))

            expect(
                screen.getByTestId('width-div-elem').textContent,
            ).toEqual('75')
        })
    })

    describe('with elementRef.current', () => {
        test('sets initial height to height value', () => {
            render(
                <ComponentUseOffsetListenerMock
                    elementRef={ {
                        current: {
                            offsetHeight: 25,
                            offsetWidth: 100,
                        },
                    } }
                />,
            )

            expect(
                screen.getByTestId('height-div-elem').textContent,
            ).toEqual('25')
        })

        test('sets initial width to width value', () => {
            render(
                <ComponentUseOffsetListenerMock
                    elementRef={ {
                        current: {
                            offsetHeight: 25,
                            offsetWidth: 100,
                        },
                    } }
                />,
            )

            expect(
                screen.getByTestId('width-div-elem').textContent,
            ).toEqual('100')
        })

        test('sets height on resize with offset height value', () => {
            const { rerender } = render(
                <ComponentUseOffsetListenerMock
                    elementRef={ {
                        current: {
                            offsetHeight: 25,
                            offsetWidth: 100,
                        },
                    } }
                />,
            )

            rerender(
                <ComponentUseOffsetListenerMock
                    elementRef={ {
                        current: {
                            offsetHeight: 50,
                            offsetWidth: 75,
                        },
                    } }
                />,
            )

            expect(
                screen.getByTestId('height-div-elem').textContent,
            ).toEqual('25')

            fireEvent(window, new Event('resize'))

            expect(
                screen.getByTestId('height-div-elem').textContent,
            ).toEqual('50')
        })

        test('sets width on resize with offset width value', () => {
            const { rerender } = render(
                <ComponentUseOffsetListenerMock
                    elementRef={ {
                        current: {
                            offsetHeight: 25,
                            offsetWidth: 100,
                        },
                    } }
                />,
            )

            rerender(
                <ComponentUseOffsetListenerMock
                    elementRef={ {
                        current: {
                            offsetHeight: 50,
                            offsetWidth: 75,
                        },
                    } }
                />,
            )

            expect(
                screen.getByTestId('width-div-elem').textContent,
            ).toEqual('100')

            fireEvent(window, new Event('resize'))

            expect(
                screen.getByTestId('width-div-elem').textContent,
            ).toEqual('75')
        })
    })
})

describe('OffsetListener snapshot', () => {
    test('matches snapshot without elementRef.current', () => {
        const { asFragment } = render(
            <ComponentUseOffsetListenerMock
                elementRef={ {} }
            />,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('matches snapshot with elementRef.current', () => {
        const { asFragment } = render(
            <ComponentUseOffsetListenerMock
                elementRef={ {
                    current: {
                        offsetHeight: 25,
                        offsetWidth: 100,
                    },
                } }
            />,
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
