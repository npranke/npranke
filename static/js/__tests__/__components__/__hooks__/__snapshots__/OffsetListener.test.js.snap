// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffsetListener snapshot matches snapshot with elementRef.current 1`] = `
<DocumentFragment>
  <div
    class="component-use-offset-listener-mock"
  >
    <div
      class="height"
      data-testid="height-div-elem"
    >
      25
    </div>
    <div
      class="width"
      data-testid="width-div-elem"
    >
      100
    </div>
  </div>
</DocumentFragment>
`;

exports[`OffsetListener snapshot matches snapshot without elementRef.current 1`] = `
<DocumentFragment>
  <div
    class="component-use-offset-listener-mock"
  >
    <div
      class="height"
      data-testid="height-div-elem"
    >
      0
    </div>
    <div
      class="width"
      data-testid="width-div-elem"
    >
      0
    </div>
  </div>
</DocumentFragment>
`;
